larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -257,106 +293,132 @@ void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const
 void LArPandoraInput::CreatePandoraReadoutGaps(const Settings &settings, const LArDriftVolumeMap &driftVolumeMap)
 {
     mf::LogDebug("LArPandora") << " *** LArPandoraInput::CreatePandoraReadoutGaps(...) *** " << std::endl;
-
+    
     if (!settings.m_pPrimaryPandora)
         throw cet::exception("LArPandora") << "CreatePandoraReadoutGaps - primary Pandora instance does not exist ";
-
+    
     const pandora::Pandora *pPandora(settings.m_pPrimaryPandora);
-
+    
     art::ServiceHandle<geo::Geometry const> theGeometry;
     const lariov::ChannelStatusProvider &channelStatus(art::ServiceHandle<lariov::ChannelStatusService const>()->GetProvider());
+    
+    const bool isDualPhase(theGeometry->MaxPlanes() == 2);
 
     for (unsigned int icstat = 0; icstat < theGeometry->Ncryostats(); ++icstat)
     {
         for (unsigned int itpc = 0; itpc < theGeometry->NTPC(icstat); ++itpc)
         {
             const geo::TPCGeo &TPC(theGeometry->TPC(itpc));
-
+            
             for (unsigned int iplane = 0; iplane < TPC.Nplanes(); ++iplane)
             {
                 const geo::PlaneGeo &plane(TPC.Plane(iplane));
                 const float halfWirePitch(0.5f * theGeometry->WirePitch(plane.View()));
-                const unsigned int nWires(theGeometry->Nwires(geo::PlaneID(icstat, itpc, plane.View())));
-
+                const unsigned int nWires(theGeometry->Nwires(plane.ID()));
```

Comment:
These three changes, lines 317, 323, 361 (in modified file) seem to be the most involved issues, as they affect SP running too. Any specific motivation for these changes, and are we completely sure that things are unchanged for the SP use cases? One way to check this might be to run up a ProtoDUNE-SP job (and/or a uboone job) and use a PandoraWriter to dump the gap geometry to XML. We can diff the XMLs obtained with this feature on and off, and that should be a good test.